src/Core/Mutation/MutationTracker.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Mutation;
+
+use Doctrine\ORM\EntityManagerInterface;
+use PrestaShop\PrestaShop\Core\Context\ApiClientContext;
+use PrestaShop\PrestaShop\Core\Context\EmployeeContext;
+use PrestaShopBundle\Entity\Mutation;
+
+/**
+ * The mutation track service helps to add mutation from any service in the code base, it automatically
+ * fills the related potential modifiers (employee and/or api client from the context). The purpose is
+ * to let it automatically set these associations, but you can manually override those values if needed,
+ * but as long as these parameters are empty the default value will always be set if present, so if you
+ * need more accurate control on this association you should create and persist the mutation manually.
+ *
+ * It also contains a few action constants as the most usually used, but the action remains a customizable
+ * field.
+ */
+class MutationTracker
+{
+    public const CREATE_ACTION = 'create';
+    public const UPDATE_ACTION = 'update';
```

Comment:
This may be a controversial proposal, but do you think we could log a real difference between the old and new states of the given row? Handling it automatically could be PITA, but what about having an old and new state passed to the MutationTracker? 🤔 

A very simple thing I do in one of the projects:

```
$changes = [
    'fieldName1' => ['old' => $oldValue1, 'new' => $newValue1],
    'fieldName2' => ['old' => $oldValue2, 'new' => $newValue2],
];
```

so it could be:

`$mutation->setChanges(json_encode($changes));`

Of course, this should be opt-in as it may drastically increase the size of the table.

Reply:
Also, we should support all types of actions - delete, change configuratuon, install module, remove module etc

Reply:
but it is not about module, there are only three actions but who performed them would be different